src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -91,6 +91,16 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
     sessionTemplateReference: String,
   ): Boolean
 
+  @Query(
+    "SELECT count(v) > 0 FROM Visit v " +
+      "WHERE v.sessionTemplateReference = :sessionTemplateReference AND " +
+      "v.visitStart >= :visitsFromDate",
+  )
+  fun hasVisitsForSessionTemplateFromDate(
+    sessionTemplateReference: String,
+    visitsFromDate: LocalDateTime,
+  ): Boolean
+
```

Comment:
`SessionTemplateRepository.getVisitCount`
I wonder if we should scrap one of these methods,...

Reply:
In both cases we use `v.visit_start` should we be using the template start and end date?

Reply:
Thanks @AledWEvans . I have removed hasVisitsForSessionTemplateFromDate and refactored hasVisitsForSessionTemplate to use validFromDate.